cocos/animation/marionette/animation-graph-context.ts

```diff
@@ -558,3 +651,68 @@ class MetaValueHandleInternal implements MetaValueHandle {
 
     private _host: AnimationGraphPoseLayoutMaintainer;
 }
+
+/**
+ * The update context for animation graph building blocks(state machine/pose node/motion...etc).
+ */
+export interface AnimationGraphUpdateContext {
+    /**
+     * Delta time to update.
+     */
+    readonly deltaTime: number;
+
+    /**
+     * Indicative weight of the updating target.
+     *
+     * The updating target shall not, for example, weight the result pose by this weight.
+     */
+    readonly indicativeWeight: number;
+}
+
+/**
+ * Utility class to generate animation graph context.
+ *
+ * The result of each method of this class is kept available until next call on any of these methods.
+ */
+export class AnimationGraphUpdateContextGenerator {
```

Comment:
I don't like this piece of shit code. But we have to since we can not allocate in updating stage.